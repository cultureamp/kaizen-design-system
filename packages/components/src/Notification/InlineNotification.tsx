import React, { HTMLAttributes } from "react"
import classnames from "classnames"
import { OverrideClassName } from "~types/OverrideClassName"
import styles from "./InlineNotification.module.scss"

/**
 * @todo: Refer to `docs/` for further code standards and guidelines
 */

/**
 * @todo: Replace `HTMLAttributes<HTMLDivElement>` with attributes/props you need to extend
 */
export type InlineNotificationProps = {
  /** @todo: Add custom props here */
  exampleRequiredString: string
  isReversed?: boolean
} & OverrideClassName<HTMLAttributes<HTMLDivElement>>

export const InlineNotification = ({
  exampleRequiredString,
  isReversed = false,
  classNameOverride,
  ...restProps
}: InlineNotificationProps): JSX.Element => {
  return (
    <div
      className={classnames(
        styles.inlineNotification,
        isReversed && styles.isReversed,
        classNameOverride,
      )}
      {...restProps}
    >
      {exampleRequiredString}
    </div>
  )
}

InlineNotification.displayName = "InlineNotification"
