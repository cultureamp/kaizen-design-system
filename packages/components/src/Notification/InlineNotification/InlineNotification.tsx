import React from "react"
import classnames from "classnames"
import { HeadingProps } from "~components/Heading"
import { OverrideClassName } from "~types/OverrideClassName"
import {
  GenericNotification,
  GenericNotificationType,
  GenericNotificationVariant,
} from "../subcomponents/GenericNotification"
import styles from "../subcomponents/GenericNotification/GenericNotification.module.scss"

export type InlineNotificationBase = OverrideClassName<{
  children?: React.ReactNode
  /** @default false */
  persistent?: boolean
  /** @default false */
  hideCloseIcon?: boolean
  onHide?: () => void
  noBottomMargin?: boolean
  forceMultiline?: boolean
  headingProps?: HeadingProps
  isSubtle?: boolean
}>

export type InlineNotificationProps = InlineNotificationBase &
  (GenericNotificationType | GenericNotificationVariant)

/**
 * {@link https://cultureamp.atlassian.net/wiki/spaces/DesignSystem/pages/3082093392/Inline+Notification Guidance} |
 * {@link https://cultureamp.design/storybook/?path=/docs/components-notifications-inline-notification--docs Storybook}
 */
export const InlineNotification = ({
  isSubtle,
  hideCloseIcon = false,
  persistent = false,
  classNameOverride,
  ...otherProps
}: InlineNotificationProps): JSX.Element => (
  <GenericNotification
    style="inline"
    persistent={persistent || hideCloseIcon}
    classNameOverride={classnames(classNameOverride, [
      isSubtle && styles.subtle,
    ])}
    {...otherProps}
  />
)

InlineNotification.displayName = "InlineNotification"
