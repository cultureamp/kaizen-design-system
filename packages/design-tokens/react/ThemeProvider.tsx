import React from "react"
import { ThemeManager, defaultTheme, Theme } from "../"

export const ThemeContext = React.createContext<Theme>(defaultTheme)

/**
 * Wrap your application in this provider using a ThemeManager, to synchronise it with a react context.
 * This allows child react elements to more easily use theme variables, using the {@link useTheme} hook.
 */
export const ThemeProvider = ({
  themeManager,
  ...props
}: {
  themeManager: ThemeManager
  children: React.ReactNode
}) => {
  const [theme, setTheme] = React.useState<Theme>(
    themeManager.getCurrentTheme()
  )
  React.useEffect(() => {
    let cancelled = false
    const listener = (newTheme: Theme) => {
      if (!cancelled) setTheme(newTheme)
    }
    themeManager.addThemeChangeListener(listener)
    return () => {
      cancelled = true
      themeManager.removeThemeChangeListener(listener)
    }
  }, [])

  return (
    <ThemeContext.Provider value={theme}>
      {props.children}
    </ThemeContext.Provider>
  )
}

export const useTheme = () => React.useContext(ThemeContext)
