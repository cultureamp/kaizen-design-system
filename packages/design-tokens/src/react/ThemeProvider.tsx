import React, { useState } from "react"
import { ThemeManager, defaultTheme, Theme } from ".."

export const ThemeContext = React.createContext<Theme>(defaultTheme)

/**
 * Wrap your application in this provider using a ThemeManager, to synchronise it with a react context.
 * This allows child react elements to more easily use theme variables, using the {@link useTheme} hook.
 */
/**
 * @deprecated ThemeProvider is deprecated. Please use KaizenProvider from "@kaizen/components" instead.
 */
export const ThemeProvider = ({
  themeManager,
  ...props
}: {
  themeManager?: ThemeManager
  children: React.ReactNode
}): JSX.Element => {
  const [themeManagerInstance] = useState(
    () => themeManager || new ThemeManager(defaultTheme)
  )

  const [theme, setTheme] = React.useState<Theme>(
    themeManagerInstance.getCurrentTheme()
  )
  React.useEffect(() => {
    let cancelled = false
    const listener = (newTheme: Theme): void => {
      if (!cancelled) setTheme(newTheme)
    }
    themeManagerInstance.addThemeChangeListener(listener)
    return () => {
      cancelled = true
      themeManagerInstance.removeThemeChangeListener(listener)
    }
  }, [])

  return (
    <>
      <ThemeContext.Provider value={theme}>
        {props.children}
      </ThemeContext.Provider>
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-bold.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-demi-bold.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-medium.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-regular.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/tiempos/tiempos-headline-bold.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
    </>
  )
}

export const useTheme = (): Theme => React.useContext(ThemeContext)
