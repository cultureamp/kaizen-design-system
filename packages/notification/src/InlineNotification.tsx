import React from "react"
import { HeadingProps } from "@kaizen/typography"
import GenericNotification, {
  NotificationType,
} from "./components/GenericNotification"
import styles from "./components/GenericNotification.module.scss"

export type InlineNotificationProps = {
  type: NotificationType
  children?: React.ReactNode
  autohide?: boolean
  autohideDelay?: "short" | "long"
  persistent: boolean
  hideCloseIcon: boolean
  onHide?: () => void
  automationId?: string
  noBottomMargin?: boolean
  forceMultiline?: boolean
  headingProps?: HeadingProps
  isSubtle?: boolean
  /**
   * **Deprecated:** Use headingProps
   * @deprecated
   */
  title?: string
}

/**
 * {@link https://cultureamp.design/components/inline-notification/ Guidance} |
 * {@link https://cultureamp.design/storybook/?path=/docs/components-notification-inline-notification--default-kaizen-demo Storybook}
 */
export const InlineNotification = ({
  persistent,
  hideCloseIcon,
  isSubtle,
  ...otherProps
}: InlineNotificationProps): JSX.Element => (
  <GenericNotification
    style="inline"
    persistent={persistent || hideCloseIcon}
    classNameOverride={isSubtle ? styles.subtle : undefined}
    {...otherProps}
  />
)

InlineNotification.defaultProps = {
  autohideDelay: "short",
  autohide: false,
  hideCloseIcon: false,
  persistent: false,
}

InlineNotification.displayName = "InlineNotification"
