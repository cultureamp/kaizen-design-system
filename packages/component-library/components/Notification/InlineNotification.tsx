import * as React from "react"
import GenericNotification, {
  NotificationType,
} from "./components/GenericNotification"

type Props = {
  type: NotificationType
  title: string
  children: React.ReactNode
  autohide?: boolean
  autohideDelay?: "short" | "long"
  persistent: boolean
  hideCloseIcon: boolean
  onHide?: () => void
  automationId?: string
  noBottomMargin?: boolean
}

const InlineNotification = ({
  persistent,
  hideCloseIcon,
  ...otherProps
}: Props) => (
  <GenericNotification
    style="inline"
    persistent={persistent || hideCloseIcon}
    {...otherProps}
  />
)

InlineNotification.defaultProps = {
  autohideDelay: "short",
  autohide: false,
  hideCloseIcon: false,
  persistent: false,
}

export default InlineNotification
