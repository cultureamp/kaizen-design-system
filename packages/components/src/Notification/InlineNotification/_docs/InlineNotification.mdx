import { Canvas, Controls, Meta, Description} from "@storybook/blocks"
import { KAIOInstallation, ResourceLinks } from "~storybook/components"
import * as InlineNotificationExamples from "./InlineNotification.stories"

<Meta title="Components/Notifications/Inline Notification" />

# Inline Notification

<ResourceLinks
  sourceCode="https://github.com/cultureamp/kaizen-design-system/tree/main/packages/components/src/Notification/InlineNotification"
  figma="https://www.figma.com/file/eZKEE5kXbEMY3lx84oz8iN/%F0%9F%92%9C-UI-Kit%3A-Heart?type=design&node-id=1929%3A21594&mode=design&t=HmSOpdHdka19ZmTO-1"
  designGuidelines="https://cultureamp.atlassian.net/wiki/spaces/DesignSystem/pages/3082093392/Inline+Notification"

/>

<KAIOInstallation exportNames="InlineNotification" />

## Overview

`<InlineNotification>` is used to display contextual information, including positive feedback or confirmation about an action, warnings about possible issues or show-stopping errors.

<Canvas of={InlineNotificationExamples.Playground} />

## API

<Controls of={InlineNotificationExamples.Playground} />


### persistent

<Description of={InlineNotificationExamples.Persistent} />
<Canvas of={InlineNotificationExamples.Persistent} />

### forceMultiline

<Description of={InlineNotificationExamples.ForceMultilineDemo} />
<Canvas of={InlineNotificationExamples.ForceMultilineDemo} />

<Description of={InlineNotificationExamples.ContentMultilineDemo} />
<Canvas of={InlineNotificationExamples.ContentMultilineDemo} />
