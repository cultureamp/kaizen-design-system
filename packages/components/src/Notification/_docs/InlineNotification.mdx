import { Canvas, Controls, Description, Meta } from "@storybook/blocks"
import { ResourceLinks, KaioNotification, Installation } from "~storybook/components"
import * as InlineNotificationStories from "./InlineNotification.stories"

<Meta of={InlineNotificationStories} />

# InlineNotification

{/*
  * @todo:
  * - Adjust Github link (if needed)
  * - Add Figma and Design Guidelines (Confluence) link
  */}
<ResourceLinks
  sourceCode="https://github.com/cultureamp/kaizen-design-system/tree/main/packages/components/src/InlineNotification"
  figma=""
  designGuidelines=""
  className="!mb-8"
/>

<KaioNotification />

<Installation
  installCommand="yarn add @kaizen/components"
  importStatement='import { InlineNotification } from "@kaizen/components"'
/>

## Overview

{/* @todo: Update summary. */}
Brief summary of the component here.

<Canvas of={InlineNotificationStories.Playground} />
<Controls of={InlineNotificationStories.Playground} />

## API

{/*
  * @todo: Describe props and provide example stories.
  * You can choose to write in MDX or to use <Description />.
  */}
### Reversed
<Description of={InlineNotificationStories.Reversed} />
<Canvas of={InlineNotificationStories.Reversed} />
