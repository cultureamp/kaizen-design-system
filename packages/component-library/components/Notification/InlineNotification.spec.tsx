import { cleanup, render } from "@testing-library/react"
import * as React from "react"
import InlineNotification from "./InlineNotification"

afterEach(cleanup)

test("The basic notification renders correctly", () => {
  const { container } = render(
    <InlineNotification type="cautionary" title="Warning">
      Something has gone wrong
    </InlineNotification>
  )

  expect(container.querySelector(".cancel")).toBeTruthy()
  expect(container.firstChild).toMatchSnapshot()
})

test("Persistent versions of the notifications work", () => {
  const { container } = render(
    <InlineNotification type="negative" title="Error" persistent>
      Something has gone wrong
    </InlineNotification>
  )

  expect(container.querySelector(".cancel")).toBeFalsy()
  expect(container.firstChild).toMatchSnapshot()
})
