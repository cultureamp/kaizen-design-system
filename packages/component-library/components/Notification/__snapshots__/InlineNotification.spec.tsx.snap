// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Persistent versions of the notifications work 1`] = `
<div
  class="notification negative inline hidden"
  data-automation-class="generic-notification negative inline"
  style="margin-top: 0px;"
>
  <div
    class="icon"
  >
    <svg
      aria-hidden="true"
      aria-labelledby="80284680-3304-5869-9788-7428e6c6ee86"
      class="icon inheritSize"
      focusable="false"
      role="presentation"
    >
      <use
        xlink:href="#undefined"
      />
    </svg>
  </div>
  <div
    class="textContainer"
  >
    <h6
      class="title"
    >
      Error
    </h6>
    <p
      class="text"
    >
      Something has gone wrong
    </p>
  </div>
</div>
`;

exports[`The basic notification renders correctly 1`] = `
<div
  class="notification cautionary inline hidden"
  data-automation-class="generic-notification cautionary inline"
  style="margin-top: 0px;"
>
  <div
    class="icon"
  >
    <svg
      aria-hidden="true"
      aria-labelledby="80284680-3304-5869-9788-7428e6c6ee86"
      class="icon inheritSize"
      focusable="false"
      role="presentation"
    >
      <use
        xlink:href="#undefined"
      />
    </svg>
  </div>
  <div
    class="textContainer"
  >
    <h6
      class="title"
    >
      Warning
    </h6>
    <p
      class="text"
    >
      Something has gone wrong
    </p>
  </div>
  <button
    class="cancel"
    type="button"
  >
    <span
      class="cancelInner"
    >
      <svg
        aria-hidden="false"
        aria-labelledby="faffea72-0fc0-5bc3-8038-936bd9f22ca8"
        class="icon"
        focusable="false"
        role="img"
      >
        <title
          id="faffea72-0fc0-5bc3-8038-936bd9f22ca8"
        >
          close notification
        </title>
        <use
          xlink:href="#undefined"
        />
      </svg>
    </span>
  </button>
</div>
`;
